### DISCUSSION TITLE
useHistory().push() or direct <Link>

### DISCUSSION THREAD

**Discussion by akramnarejo** [remix-run/react-router#8185] [Q&A] [answered]
How it is different to use `useHistory().push('/')` method? when we can do the same by providing the `<Link to='/' />` directly.

**Comment by Svish** [comment, marked as answer]
Links are for links the user click on. `push` is for when the navigation needs to happen programmatically. It could for example be used to redirect a user to a login-form if they need to authenticate, or when a user successfully posts a new blog post to automatically redirect the user to that blog-post, or back to the list of blog posts, or something like that.
